libcaf_io/src/default_multiplexer.cpp

```diff
@@ -1339,35 +1349,33 @@ expected<void> new_ip_acceptor_impl(native_socket fd, uint16_t port,
   CALL_CFUN(res, cc_zero, "bind",
             bind(fd, reinterpret_cast<sockaddr*>(&sa),
                  static_cast<socklen_t>(sizeof(sa))));
-  return unit;
+  return sguard.release();
 }
 
 expected<native_socket> new_tcp_acceptor_impl(uint16_t port, const char* addr,
                                               bool reuse_addr) {
   CAF_LOG_TRACE(CAF_ARG(port) << ", addr = " << (addr ? addr : "nullptr"));
-  protocol proto = ipv6;
-  if (addr != nullptr) {
-    auto addrs = interfaces::native_address(addr);
-    if (!addrs)
-      return make_error(sec::cannot_open_port, "Invalid ADDR", addr);
-    proto = addrs->second;
-    CAF_ASSERT(proto == ipv4 || proto == ipv6);
-  }
-  CALL_CFUN(fd, cc_valid_socket, "socket",
-            socket(proto == ipv4 ? AF_INET : AF_INET6, SOCK_STREAM, 0));
-  // sguard closes the socket in case of exception
-  socket_guard sguard(fd);
-  if (reuse_addr) {
-    int on = 1;
-    CALL_CFUN(tmp1, cc_zero, "setsockopt",
-              setsockopt(fd, SOL_SOCKET, SO_REUSEADDR,
-                         reinterpret_cast<setsockopt_ptr>(&on),
-                         static_cast<socklen_t>(sizeof(on))));
+  auto addrs = interfaces::server_address(port, addr);
+  if (addrs.empty())
+    return make_error(sec::cannot_open_port, "No local interface available",
+                      addr);
+  int fd = invalid_native_socket;
+  for (auto& elem : addrs) {
+    const char* addr = elem.first.c_str();
+    auto p = elem.second == ipv4
+           ? new_ip_acceptor_impl<AF_INET>(port, addr, reuse_addr)
+           : new_ip_acceptor_impl<AF_INET6>(port, addr, reuse_addr);
+    if (!p) {
+      CAF_LOG_ERROR(p.error());
+      continue;
+    }
+    fd = *p;
+    break;
   }
-  auto p = proto == ipv4 ? new_ip_acceptor_impl<AF_INET>(fd, port, addr)
-                         : new_ip_acceptor_impl<AF_INET6>(fd, port, addr);
-  if (!p)
-    return std::move(p.error());
+  if (fd == invalid_native_socket)
+    return make_error(sec::cannot_open_port,
```

Comment:
This is where I would put the log output. However, this is not a hard error in the program. Either of `CAF_LOG_WARNING` or `CAF_LOG_INFO` is a better fit.